build.gradle

```diff
@@ -343,42 +333,26 @@ task setupEclipse {
     group "Setup"
     dependsOn eclipseProject, createEclipseLaunches, resetEclipseDeps
 }
+*/
 
 // RUN SERVER TASKS
 
 compileJava.options.encoding = "UTF-8"
 compileTestJava.options.encoding = "UTF-8"
 
 appengine {
-    httpPort = 8888
-    downloadSdk = true
-    jvmFlags = ["-Xss2m", "-Dfile.encoding=UTF-8",
-                // absolute paths are not supported, the following is relative to the exploded war directory
-                // this only specifies the datastore path, but the indexes are still generated in
-                // WEB-INF/appengine-generated.
-                "-Ddatastore.backing_store=../../local_db.bin"]
-    appcfg {
-        oauth2 = true
+    run {
+        port = 8888
+        jvmFlags = ["-Xss2m", "-Dfile.encoding=UTF-8"]
     }
-}
-
-appengineRun {
-    if (project.hasProperty("disable_daemon")) {
-        daemon = false
-    } else {
-        daemon = true
+    deploy {
+        project = 'teammates-john'
+        version = '4-18'
```

Comment:
These need to be read from system properties instead so they can be set via command line. Otherwise, deployment would always involve editing this file, deploying, then discarding the changes to avoid them showing up in version control.

> There are two different places to configure app ID and app version; build.gradle and appengine-web.xml. Deployment makes use of the former while dev server makes use of the latter. Not a loss of function; just some inconvenience. Hopefully it will be converted back to single configuration point some time in the future.

Why not remove it from `appengine-web.template.xml`? I remember reading Google's docs that sometime in the near future the `application` and `version` tags will be considered invalid such that they won't even let you start the dev server if those tags are present in the xml.

Reply:
Tried just now; removing `<application>` works, but removing `<version>` will break some admin features, hopefully I can resolve it quick.

Reply:
Will you be resolving it in this PR?

Reply:
As it turns out, our logs system is dependent on the version number being set, and this includes having to set it in two different places for now until Google decides otherwise. In that situation, I'm afraid I can't do much here. But I can remove the `<application>` tag, if you wish so.

Reply:
Hmm it seems they've changed their stance and those tags should indeed remain in `appengine-web.xml` (I remember in a previous version of Cloud SDK, starting the dev server with those tags present in the XML would result in a warning, but that seems to have disappeared and now, as you said, it's the only way to set the version for the dev server).

Can we make gradle read the values from the XML? This works:

```groovy
    deploy {
        Node appengineWebXml = new XmlParser().parse('src/main/webapp/WEB-INF/appengine-web.xml')
        project = appengineWebXml.application.text()
        version = appengineWebXml.version.text()
    }
```